services/anvil/fork_test.go

```diff
@@ -0,0 +1,47 @@
+package anvil
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/ethereum-optimism/optimism/op-bindings/bindings"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind/backends"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/params"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetL2BlockNumberAtSpecificL1Block(t *testing.T) {
+	// Setup
+
+	privateKey, err := crypto.GenerateKey()
+	auth := bind.NewKeyedTransactor(privateKey)
+	auth.GasLimit = 8000000
+	from := crypto.PubkeyToAddress(privateKey.PublicKey)
+	backend := backends.NewSimulatedBackend(core.GenesisAlloc{from: {Balance: big.NewInt(params.Ether)}}, 50_000_000)
+	opts, err := bind.NewKeyedTransactorWithChainID(privateKey, big.NewInt(1337))
+	defer backend.Close()
+
+	l2OutputOracleAddress, _, _, err := bindings.DeployL2OutputOracle(
+		opts,
+		backend,
+		big.NewInt(10),
+		big.NewInt(2),
+		big.NewInt(100),
+	)
+	require.NoError(t, err)
+
+	backend.Commit()
+
+	l1BlockNumber := big.NewInt(1) // As an example
+	l2BlockNumber, err := GetL2BlockNumberAtSpecificL1Block(l1BlockNumber, backend, l2OutputOracleAddress)
+	require.NoError(t, err)
+
+	// TODO add proposals
+
+	expectedL2BlockNumber := big.NewInt(42)
+	require.Equal(t, expectedL2BlockNumber, l2BlockNumber)
```

Comment:
```suggestion
	// expectedL2BlockNumber := big.NewInt(42)
	// require.Equal(t, expectedL2BlockNumber, l2BlockNumber)
```
Will finish this test after the proposer is added to mocktimism